cogs/welcome/welcome.py

```diff
@@ -318,6 +341,71 @@ async def logServerLeave(self, leaveUser: discord.Member):
                     leaveUser.id,
                 )
 
+    async def generateRandWelcomeImg(self, user, guild):
+        """creates an image for the specific player using their avatar and an image from the random image pool, then returns it"""
+        base = Image.open(
+            self.imgDir
+            / str(guild.id)
+            / random.choice(os.listdir(os.path.join(self.imgDir, str(guild.id))))
+        )
+        mask = Image.open(os.path.join(data_manager.bundled_data_path(self), "MASK.png"))
+        borderOverlay = Image.open(
+            os.path.join(data_manager.bundled_data_path(self), "BORDER.png")
+        )
+        borderOverlayMask = Image.open(
+            os.path.join(data_manager.bundled_data_path(self), "BORDER_mask.png")
+        )
+        # get avatar from User
+        avatar: bytes
+        session = aiohttp.ClientSession()
+        # a header to successfully download user avatars for use
+        usedHeader = {"User-agent": "Mozilla/5.0"}
+
+        try:
+            async with session.get(str(user.avatar.url), headers=usedHeader) as webp:
+                avatar = await webp.read()
+        except aiohttp.ClientResponseError:
+            pass
+
+        with Image.open(io.BytesIO(avatar)) as retrieved_avatar:
+            if not retrieved_avatar:
+                base.close()
+                mask.close()
+                borderOverlay.close()
+                borderOverlayMask.close()
+                return
+            else:
+                retrieved_avatar = retrieved_avatar.resize((325, 325), 1)
+                base.paste(borderOverlay, (434, 0), borderOverlayMask)
+                base.paste(retrieved_avatar, (434, 0), mask)
+                generated = io.BytesIO()
+                base.save(generated, format="png")
+                generated.seek(0)
+                base.close()
+                mask.close()
+                borderOverlay.close()
+                borderOverlayMask.close()
+                return generated
+
+    async def ensureCurrentServerHasImgCache(self, channel):
+        """
+        given a channel, checks if there is a folder in the image cache for the associated server. If one doesn't exist, creates it.
+        """
+        idStr = str(channel.guild.id)
+        fp = os.path.join(self.imgDir, idStr)
+        if os.path.exists(fp):
+            return
+
+        try:
+            os.makedirs(fp, exist_ok=True)
+            await channel.send("No image cache folder found for this server! Created one")
+
+        except OSError as info:
+            LOGGER.info(
+                "No folder for given server ID found. Created folder for server: " + idStr,
```

Comment:
Please use %s instead of concatenating: i.e. `"...folder for server: %s", idStr, ...`